File: src/lib/runners/custom/cpp/remove-duplicates-from-sorted-array.cpp
#include <iostream>
#include <vector>
#include <string>
#include <queue>
#include <sstream>
#include <climits>
#include <unordered_map>
#include <unordered_set>
#include <set>
#include <map>
#include <algorithm>
#include <numeric>
#include <cmath>
#include <stack>

using namespace std;

// --- Manual Parsers ---
string trim(const string& s) {
    size_t start = s.find_first_not_of(" \t\r\n");
    if (start == string::npos) return "";
    return s.substr(start, s.find_last_not_of(" \t\r\n") - start + 1);
}

int parseInt(string s) { return stoi(trim(s)); }

vector<string> splitArray(string s) {
    vector<string> res;
    s = trim(s);
    if(s.size() >= 2 && s.front() == '[' && s.back() == ']') s = s.substr(1, s.size()-2);
    int depth = 0;
    string cur;
    for(char c : s) {
        if(c == '[') depth++;
        else if(c == ']') depth--;
        if(c == ',' && depth == 0) {
            res.push_back(cur);
            cur = "";
        } else {
            cur += c;
        }
    }
    if(!cur.empty()) res.push_back(cur);
    return res;
}

vector<int> parseIntArray(string s) {
    vector<int> res;
    for(auto& x : splitArray(s)) res.push_back(parseInt(x));
    return res;
}

vector<string> parseInputString(string input) {
    vector<string> result;
    int depth = 0;
    string current;
    for (size_t i = 0; i < input.size(); i++) {
        char ch = input[i];
        if (ch == '[' || ch == '(' || ch == '{') depth++;
        else if (ch == ']' || ch == ')' || ch == '}') depth--;
        current += ch;
        if ((ch == ',' && depth == 0) || i == input.size() - 1) {
            if (ch == ',') current.pop_back();
            size_t eq = current.find('=');
            if (eq != string::npos) {
                result.push_back(trim(current.substr(eq + 1)));
            }
            current.clear();
        }
    }
    return result;
}

// --- User's Solution Code ---
${userCode}

// --- Main Execution Loop ---
int main() {
    string input_data((istreambuf_iterator<char>(cin)), istreambuf_iterator<char>());
    
    string search_key = "\"input\"";
    size_t pos = 0;
    while ((pos = input_data.find(search_key, pos)) != string::npos) {
        pos += search_key.length();
        
        size_t colon_pos = input_data.find(':', pos);
        if (colon_pos == string::npos) break;
        
        size_t quote_start = input_data.find('"', colon_pos);
        if (quote_start == string::npos) break;
        
        size_t quote_end = quote_start + 1;
        while (quote_end < input_data.length()) {
            if (input_data[quote_end] == '"' && input_data[quote_end - 1] != '\\') {
                break;
            }
            quote_end++;
        }
        
        if (quote_end >= input_data.length()) break;
        
        string tc_input = input_data.substr(quote_start + 1, quote_end - quote_start - 1);
        
        string unescaped;
        for (size_t i = 0; i < tc_input.length(); i++) {
            if (tc_input[i] == '\\' && i + 1 < tc_input.length() && tc_input[i+1] == '"') {
                unescaped += '"';
                i++;
            } else {
                unescaped += tc_input[i];
            }
        }
        tc_input = unescaped;
        
        // Now find the "expected" field for this testcase
        string exp_search_key = "\"expected\"";
        size_t exp_pos = input_data.find(exp_search_key, quote_end);
        string exp_str = "";
        if (exp_pos != string::npos) {
            exp_pos += exp_search_key.length();
            size_t exp_colon_pos = input_data.find(':', exp_pos);
            size_t exp_quote_start = input_data.find('"', exp_colon_pos);
            size_t exp_quote_end = exp_quote_start + 1;
            while (exp_quote_end < input_data.length()) {
                if (input_data[exp_quote_end] == '"' && input_data[exp_quote_end - 1] != '\\') break;
                exp_quote_end++;
            }
            exp_str = input_data.substr(exp_quote_start + 1, exp_quote_end - exp_quote_start - 1);
            
            string exp_unescaped;
            for (size_t i = 0; i < exp_str.length(); i++) {
                if (exp_str[i] == '\\' && i + 1 < exp_str.length() && (exp_str[i+1] == '"' || exp_str[i+1] == 'n')) {
                    if (exp_str[i+1] == 'n') exp_unescaped += '\n';
                    else exp_unescaped += '"';
                    i++;
                } else {
                    exp_unescaped += exp_str[i];
                }
            }
            exp_str = exp_unescaped;
        }

        try {
            vector<string> parsed_args = parseInputString(tc_input);
            vector<int> arg0 = parseIntArray(parsed_args[0]);
            
            int orig_size = arg0.size();
            
            // Calculate expected array (unique elements)
            vector<int> expected_arr;
            if (orig_size > 0) {
                expected_arr.push_back(arg0[0]);
                for (int i = 1; i < orig_size; i++) {
                    if (arg0[i] != arg0[i-1]) expected_arr.push_back(arg0[i]);
                }
            }
            int expected_k = expected_arr.size();
            
            Solution sol;
            int k = sol.removeDuplicates(arg0);
            
            bool valid = true;
            if (k != expected_k) valid = false;
            for (int i = 0; i < k && valid; i++) {
                if (i >= (int)expected_arr.size() || arg0[i] != expected_arr[i]) valid = false;
            }
            
            if (valid) {
                cout << exp_str << endl; // Just print expected to pass the string matching
            } else {
                cout << k << ", nums = [";
                for(int i=0; i<orig_size; i++) {
                    if(i < k) {
                        cout << arg0[i];
                    } else {
                        cout << "_";
                    }
                    if(i < orig_size - 1) cout << ",";
                }
                cout << "]" << endl;
            }
            
        } catch (const exception& e) {
            cout << "{\"error\":\"" << e.what() << "\"}" << endl;
        }
        
        pos = quote_end + 1;
    }
    return 0;
}

File: src/lib/runners/custom/python/remove-duplicates-from-sorted-array.py
import sys
import json
import re
import time

def parse_input_string(input_str):
    parts = []
    depth = 0
    current = ""
    for ch in input_str:
        if ch in "([{":
            depth += 1
        elif ch in ")]}":
            depth -= 1
        current += ch
        if ch == "," and depth == 0:
            parts.append(current[:-1].strip())
            current = ""
    if current.strip():
        parts.append(current.strip())
    
    values = []
    for part in parts:
        eq_idx = part.index("=")
        val_str = part[eq_idx + 1:].strip()
        val_str = val_str.replace("null", "None").replace("true", "True").replace("false", "False")
        values.append(eval(val_str))
    return values

from typing import List, Optional

${userCode}

if __name__ == "__main__":
    input_data = sys.stdin.read()
    sol = Solution()
    total_execution_time_ms = 0

    search_key = '"input"'
    pos = 0
    while True:
        pos = input_data.find(search_key, pos)
        if pos == -1:
            break
        pos += len(search_key)
        
        colon_pos = input_data.find(':', pos)
        if colon_pos == -1: break
            
        quote_start = input_data.find('"', colon_pos)
        if quote_start == -1: break
            
        quote_end = quote_start + 1
        while quote_end < len(input_data):
            if input_data[quote_end] == '"' and input_data[quote_end - 1] != '\\': break
            quote_end += 1
            
        if quote_end >= len(input_data): break
            
        tc_input = input_data[quote_start + 1 : quote_end]
        tc_input = tc_input.replace('\\"', '"')
        
        # Find expected
        exp_search_key = '"expected"'
        exp_pos = input_data.find(exp_search_key, quote_end)
        exp_str = ""
        if exp_pos != -1:
            exp_pos += len(exp_search_key)
            exp_colon_pos = input_data.find(':', exp_pos)
            if exp_colon_pos != -1:
                exp_quote_start = input_data.find('"', exp_colon_pos)
                if exp_quote_start != -1:
                    exp_quote_end = exp_quote_start + 1
                    while exp_quote_end < len(input_data):
                        if input_data[exp_quote_end] == '"' and input_data[exp_quote_end - 1] != '\\': break
                        exp_quote_end += 1
                    if exp_quote_end < len(input_data):
                        exp_str = input_data[exp_quote_start + 1 : exp_quote_end]
                        exp_str = exp_str.replace('\\"', '"').replace('\\n', '\n')

        try:
            parsed_args = parse_input_string(tc_input)
            arg0 = parsed_args[0]
            
            # calculate expected_k
            expected_k = 0
            if len(arg0) > 0:
                expected_k = 1
                for i in range(1, len(arg0)):
                    if arg0[i] != arg0[i-1]:
                        expected_k += 1
            
            orig_size = len(arg0)
            
            start_time = time.perf_counter()
            k = sol.removeDuplicates(arg0)
            end_time = time.perf_counter()
            total_execution_time_ms += (end_time - start_time) * 1000
            
            if total_execution_time_ms > 200000:
                print('{"error":"Time Limit Exceeded"}')
                break
                
            valid = True
            if k != expected_k:
                valid = False
                
            for i in range(1, k):
                if i < len(arg0) and arg0[i] <= arg0[i-1]:
                    valid = False
            
            if valid:
                print(exp_str)
            else:
                formatted = str(k) + ", nums = ["
                for i in range(orig_size):
                    if i < k and i < len(arg0):
                        formatted += str(arg0[i])
                    else:
                        formatted += "_"
                    if i < orig_size - 1:
                        formatted += ","
                formatted += "]"
                print(formatted)
                
        except Exception as e:
            print(json.dumps({"error": str(e)}, separators=(',', ':')))
            
        pos = quote_end + 1
